### ISSUE TITLE
react-dialog: semantic updates

### ISSUE THREAD

**Issue by smhigley** [microsoft/fluentui#25151]
(discussed during the manual a11y screen reader test pass)

We should make the following two semantic updates to dialog:

1. Remove `aria-haspopup` from the trigger

This fixes the following bugs:
- JAWS switches to application mode every time it lands on a dialog trigger
- multiple windows screen readers treat it as a menu button
- `aria-haspopup=dialog` was never intended for this use case; it was for comboboxes that open dialog popups, and even that doesn't work. There has been discussion about removing `aria-haspopup` wholesale from the spec because of its issues 🤪.

2. Do not set `aria-describedby` on the dialog by default, but add a docs note for authors to add it for short confirmation-style dialogs

It ends up being extremely verbose for any dialogs that have more than one line of content. This is especially noticeable for NVDA (since NVDA reads dialog information twice by default), and for JAWS + the Alert dialog, since this causes JAWS to read the dialog body text three times. Narrator and Talkback fully ignore the attribute.

Even for screen readers only reading the description once (VoiceOver, and JAWS for non-alert dialogs), it's quite verbose for longer dialogs. I believe it'd be a safer default to remove it, and have strong up-front advice on setting it for specifically confirmation dialogs, with an example (or multiple examples) that does so.

**Comment by jurokapsiar** [comment]
@smhigley is there any other possibility to indicate that a button opens a dialog? Usually the label of the button might not suggest that and `aria-haspopup` indicates, that clicking on the button does not yet execute the action but only opens the dialog.

**Comment by smhigley** [comment]
@jurokapsiar I'd suggest `aria-expanded=false` to indicate that it opens something. I wouldn't worry about specifically indicating that it will open a dialog (vs. another type of popup), since sighted users don't get that information either, and `aria-haspopup` also doesn't communicated it reliably.

**Comment by jurokapsiar** [comment]
I agree - can you please modify the issue description and include aria-expanded?
